packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -17,6 +17,33 @@ import 'sliver.dart';
 import 'viewport.dart';
 import 'viewport_offset.dart';
 
+/// A callback function that's invoked when the [SliverAppBar] is pulled into an
+/// overscroll and stretched to the offset specified by `stretchTriggerOffset`.
+/// Must return a [Future].
+typedef StretchCallback = Future<void> Function();
```

Comment:
There are some existing types that are equivalent to this.  I'm not sure if it's better to consolidate or keep them separate, I haven't noticed that in the style guide.

basic_types.dart AsyncCallback
refresh_indicator.dart RefreshCallback
refresh.dart RefreshCallback

Reply:
Using the AsyncCallback sounds like a great idea. Let me factor out the StretchCallback.